src/services/wallets/wallet.ts

```diff
@@ -0,0 +1,72 @@
+import { HexString } from "../../utilities/types";
+
+import * as torus from "./torus/torus";
+import * as metamask from "./metamask/metamask";
+import Web3 from "web3";
+
+export enum WalletType {
+  TORUS = "TORUS",
+  METAMASK = "METAMASK",
+  // Add new WalletTypes Here
+}
+
+interface Wallet {
+  login: () => Promise<HexString>;
+  logout: () => void;
+  getAddress: () => Promise<HexString>;
+  getBalance: (walletAddress: HexString) => Promise<number>;
+  getWeb3: () => Web3;
+}
+
+export const wallet = (walletType: WalletType): Wallet => {
+  switch (walletType) {
+    case WalletType.TORUS:
+      return torusWallet;
+    case WalletType.METAMASK:
+      return metamaskWallet;
+    // Add new WalletTypes Here
+  }
+};
+
+const torusWallet: Wallet = {
```

Comment:
I might shift this out to a different file perhaps `./metamask/index.ts`. That file would then export default just the wallet. Then everything else metamask is hidden inside the ./metamask module folder.